Source/pack.cpp

```diff
@@ -152,7 +148,7 @@ void UnPackItem(const ItemPack *is, Item *id, bool isHellfire)
 		idx = RemapItemIdxFromDiablo(idx);
 	}
 
-	if (!IsItemAvailable(idx)) {
+	if (idx == IDI_NONE || idx > IDI_LAST || !IsItemAvailable(idx)) {

```

Comment:
```suggestion
	if (idx < 0 || idx > IDI_LAST || !IsItemAvailable(idx)) {
```

Reply:
what about <= IDI_NONE? : P

Reply:
Though about it, but it doesn't really make sens given its context

Reply:
From what I could see `IsItemAvailable` is used mostly in the context of a loop over `AllItemsList` so the index (ID) is assumed to be within the range of 0-166 (`AllItemsList[167]` is a marker for the end of the list).

This and the use in `RemoveInvalidItem` break that assumption. I'd like to see this check moved into the function as well since to me it makes no sense than an invalid item is available.